### PR TITLE
refined diagnosis key api to be more sensible with padded data

### PR DESCRIPTION
**Pull request by solita-alperttiti**
Some tuning to make rest-api more sensible.

### CODE REVIEW

File: exposure-notification/README.md [hunk 1]
```diff
@@ -275,8 +276,8 @@ Status API is a one-call replacement for diagnosis key current & list fetches as
           "transmissionRiskLevel": 5,
           "rollingStartIntervalNumber":  123456,
           "rollingPeriod": 144,
-          "visitedCountries": "DE",
```
Comment (jyrkija) [inline exposure-notification/README.md:278]: Why are these here for every key? Shouldn't they be once for the whole body?

File: exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/diagnosiskey/DiagnosisKeyService.java [hunk 2]
```diff
@@ -66,7 +66,7 @@ private long getExportedKeyCount(List<TemporaryExposureKey> keys) {
                 getRiskBucket(symptomsOnset, utcDateOf10MinInterval(requestKey.rollingStartIntervalNumber)),
                 requestKey.rollingStartIntervalNumber,
                 requestKey.rollingPeriod,
-                requestKey.visitedCountries,
+                requestKey.visitedCountries.keySet(),
```
Comment (jyrkija) [inline exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/diagnosiskey/DiagnosisKeyService.java]: I expected to see the filtering of value != 1 here. Where is it?

Comment (jyrkija) [reply exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/diagnosiskey/DiagnosisKeyService.java]: Oh, right below. The field inside the object might as well be the set directly; if you're filtering in the parsing part (which is fine), the map structure is pointless after that.

File: exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/diagnosiskey/Validation.java [hunk 3]
```diff
@@ -80,6 +81,14 @@ public static int validateRollingPeriod(int rollingPeriod) {
         return codes.stream().filter(code -> !code.equals("FI")).filter(Validation::validateISOCountryCode).collect(Collectors.toSet());
     }
 
+    public static Map<String, Boolean> validateISOCountryCodesWithoutFI(Map<String, Boolean> codes) {
+        return validateISOCountryCodesWithoutFI(codes.entrySet().stream().filter(Map.Entry::getValue)
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toSet()))
```
Comment (jyrkija) [inline exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/diagnosiskey/Validation.java]: One level of processing ends here with the ending ) but the indentation remains the same. This could be made clearer with some well-placed linebreaks and indentation.

File: exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/diagnosiskey/v1/TemporaryExposureKeyRequest.java [hunk 4]
```diff
@@ -15,31 +17,44 @@
...
+     * Increments of 10 minutes describing how long a key is valid
+     **/
     public final int rollingPeriod;
-    /** Set of visited countries in ISO-3166 alpha-2 format **/
-    public final Set<String> visitedCountries;
-    /** Consent to share data with efgs **/
+    /**
+     * Set of visited countries in ISO-3166 alpha-2 format
+     **/
+    public final Map<String, Boolean> visitedCountries;
```
Comment (jyrkija) [inline exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/diagnosiskey/v1/TemporaryExposureKeyRequest.java]: Move to request-level, not key-level, as they're given just once.

**Review by jyrkija — CHANGES_REQUESTED** [review]

File: exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/diagnosiskey/DiagnosisKeyService.java (hunk 2 above)
Comment (solita-alperttiti) [reply exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/diagnosiskey/DiagnosisKeyService.java]: It's complicates e.g. tests, so that's why this is it in this way.

File: exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/diagnosiskey/v1/TemporaryExposureKeyRequest.java (hunk 4 above)
Comment (solita-alperttiti) [reply exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/diagnosiskey/v1/TemporaryExposureKeyRequest.java]: Ah, you are correct. It was already there, so didn't double check think that. Good point. I'll fix this.

File: exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/diagnosiskey/Validation.java (hunk 3 above)
Comment (solita-alperttiti) [reply exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/diagnosiskey/Validation.java]: For indents I'm using ideas defaults, it might be a good idea to make some project specific settings in some point, but not know. I'll let what I can do for this.

File: exposure-notification/README.md (hunk 1 above)
Comment (solita-alperttiti) [reply exposure-notification/README.md:278]: Yes, it should.

**Review by jyrkija — APPROVED** [review]
